### PR TITLE
Avoid serializing local object

### LINKED ISSUE #321: AttributeError: Can't pickle local object 'ModelSpec.pre_processing.<locals>.<lambda>'
The code:
``` python
import torchvision
import mlflow
import rikai
import getpass
import pathlib
from example import spark

mlflow_tracking_uri = "sqlite:///mlruns.db"

mlflow.set_tracking_uri(mlflow_tracking_uri)

with mlflow.start_run():
    ####
    # Part 1: Train the model and register it on MLflow
    ####

    # Using the pretrained here to simplify the example code
    resnet_model = torchvision.models.detection.fasterrcnn_resnet50_fpn(
        pretrained=True,
        progress=False,
    )

    schema = "struct<boxes:array<array<float>>, scores:array<float>, labels:array<int>>"

    # Rikai's logger adds output_schema, pre_pocessing, and post_processing as additional
    # arguments and automatically adds the flavor / rikai model spec version
    registered_model_name = f"{getpass.getuser()}_resnet_model"
    rikai.mlflow.pytorch.log_model(
        resnet_model,
        "model",
        schema,
        registered_model_name = registered_model_name)

    ####
    # Part 2: create the model using the registered MLflow uri
    ####
    spark.conf.set("rikai.sql.ml.registry.mlflow.tracking_uri", mlflow_tracking_uri)
    spark.sql(f"""
    CREATE MODEL mlflow_resnet_m USING 'mlflow:///{registered_model_name}';
    """)

    ####
    # Part 3: predict using the registered Rikai model
    ####
    spark.sql("show models").show(1, vertical=False, truncate=False)

    work_dir = pathlib.Path().absolute()
    result = spark.sql(f"""
    select ML_PREDICT(mlflow_resnet_m, '{work_dir}/img/lena.png')
    """)

    result.printSchema()
    result.show(1, vertical=False, truncate=False)
```

The stacktrace:
```
Traceback (most recent call last):
  File "resnet_mlflow.py", line 53, in <module>
    result.show(1, vertical=False, truncate=False)
  File "/Users/da/.pyenv/versions/rikai-example/lib/python3.8/site-packages/pyspark/sql/dataframe.py", line 486, in show
    print(self._jdf.showString(n, int(truncate), vertical))
  File "/Users/da/.pyenv/versions/rikai-example/lib/python3.8/site-packages/py4j/java_gateway.py", line 1304, in __call__
    return_value = get_return_value(
  File "/Users/da/.pyenv/versions/rikai-example/lib/python3.8/site-packages/pyspark/sql/utils.py", line 117, in deco
    raise converted from None
pyspark.sql.utils.PythonException:
  An exception was thrown from the Python worker. Please see the stack trace below.
Traceback (most recent call last):
  File "/Users/da/.pyenv/versions/3.8.10/envs/rikai-example/lib/python3.8/site-packages/pyspark/python/lib/pyspark.zip/pyspark/worker.py", line 604, in main
    process()
  File "/Users/da/.pyenv/versions/3.8.10/envs/rikai-example/lib/python3.8/site-packages/pyspark/python/lib/pyspark.zip/pyspark/worker.py", line 596, in process
    serializer.dump_stream(out_iter, outfile)
  File "/Users/da/.pyenv/versions/3.8.10/envs/rikai-example/lib/python3.8/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/pandas/serializers.py", line 273, in dump_stream
    return ArrowStreamSerializer.dump_stream(self, init_stream_yield_batches(), stream)
  File "/Users/da/.pyenv/versions/3.8.10/envs/rikai-example/lib/python3.8/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/pandas/serializers.py", line 81, in dump_stream
    for batch in iterator:
  File "/Users/da/.pyenv/versions/3.8.10/envs/rikai-example/lib/python3.8/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/pandas/serializers.py", line 266, in init_stream_yield_batches
    for series in iterator:
  File "/Users/da/.pyenv/versions/3.8.10/envs/rikai-example/lib/python3.8/site-packages/pyspark/python/lib/pyspark.zip/pyspark/worker.py", line 356, in func
    for result_batch, result_type in result_iter:
  File "/Users/da/.pyenv/versions/rikai-example/lib/python3.8/site-packages/rikai/spark/sql/codegen/pytorch.py", line 69, in torch_inference_udf
    for batch in DataLoader(
  File "/Users/da/.pyenv/versions/rikai-example/lib/python3.8/site-packages/torch/utils/data/dataloader.py", line 355, in __iter__
    return self._get_iterator()
  File "/Users/da/.pyenv/versions/rikai-example/lib/python3.8/site-packages/torch/utils/data/dataloader.py", line 301, in _get_iterator
    return _MultiProcessingDataLoaderIter(self)
  File "/Users/da/.pyenv/versions/rikai-example/lib/python3.8/site-packages/torch/utils/data/dataloader.py", line 914, in __init__
    w.start()
  File "/Users/da/.pyenv/versions/3.8.10/lib/python3.8/multiprocessing/process.py", line 121, in start
    self._popen = self._Popen(self)
  File "/Users/da/.pyenv/versions/3.8.10/lib/python3.8/multiprocessing/context.py", line 224, in _Popen
    return _default_context.get_context().Process._Popen(process_obj)
  File "/Users/da/.pyenv/versions/3.8.10/lib/python3.8/multiprocessing/context.py", line 284, in _Popen
    return Popen(process_obj)
  File "/Users/da/.pyenv/versions/3.8.10/lib/python3.8/multiprocessing/popen_spawn_posix.py", line 32, in __init__
    super().__init__(process_obj)
  File "/Users/da/.pyenv/versions/3.8.10/lib/python3.8/multiprocessing/popen_fork.py", line 19, in __init__
    self._launch(process_obj)
  File "/Users/da/.pyenv/versions/3.8.10/lib/python3.8/multiprocessing/popen_spawn_posix.py", line 47, in _launch
    reduction.dump(process_obj, fp)
  File "/Users/da/.pyenv/versions/3.8.10/lib/python3.8/multiprocessing/reduction.py", line 60, in dump
    ForkingPickler(file, protocol).dump(obj)
AttributeError: Can't pickle local object 'ModelSpec.pre_processing.<locals>.<lambda>'
```

### PR DESCRIPTION
**Pull request by da-tubi**
Fix #321

### CODE REVIEW

**Comment by eddyxu** [comment]
Makes sense, could we add a test against the lambda serialization issue to prevent we introduced the same bug again?

**Comment by da-tubi** [comment]
> Makes sense, could we add a test against the lambda serialization issue to prevent we introduced the same bug again?

Yes. Unit tests added now!

File: python/tests/spark/sql/codegen/test_fs.py [hunk 1]
```diff
@@ -16,6 +16,9 @@
 import uuid
 from pathlib import Path
 from typing import Any, Dict
+import pandas as pd
```
Comment (eddyxu) [inline python/tests/spark/sql/codegen/test_fs.py:19]: please run isort

File: python/rikai/spark/sql/codegen/base.py [hunk 2]
```diff
@@ -146,7 +150,7 @@ def pre_processing(self) -> Optional[Callable]:
             or self._spec["transforms"]["pre"] is None
         ):
             # Passthrough
-            return lambda x: x
+            return _identity
```
Comment (eddyxu) [inline python/rikai/spark/sql/codegen/base.py:153]: Nice fix.

**Review by eddyxu — APPROVED** [review]
Small format nits. LGTM overall

**Pull request merged**